kedro/framework/cli/project.py

```diff
@@ -218,17 +206,53 @@ def build_docs(metadata: ProjectMetadata, open_docs):
 
 
 @forward_command(project_group, name="build-reqs")
+@click.option(
+    "--input-file",
+    "input_file",
+    multiple=False,
+    help=INPUT_FILE_HELP,
+)
+@click.option(
+    "--output-file",
+    "output_file",
+    multiple=False,
+    help=OUTPUT_FILE_HELP,
+)
 @click.pass_obj  # this will pass the metadata as first argument
 def build_reqs(
-    metadata: ProjectMetadata, args, **kwargs
+    metadata: ProjectMetadata, input_file, output_file, args, **kwargs
 ):  # pylint: disable=unused-argument
-    """Build the project dependency requirements."""
+    """Run `pip-compile` on requirements.txt or the user defined input file and save
+    the compiled requirements to requirements.lock or the user defined output file.
+    """
+
     source_path = metadata.source_dir
-    _build_reqs(source_path, args)
+    input_file = input_file or source_path / "requirements.txt"
+    output_file = output_file or source_path / "requirements.lock"
+
+    if input_file.is_file():
```

Comment:
```suggestion
    if Path(input_file).is_file():
```
so that it works when user provides an `input_file` (which is a `str`).

Alternatively we can cast them earlier, L230-231, and revert back to `input_file.name` in the error message.

Reply:
I opted for casting earlier in L230-231